gapic/src/main/com/google/gapid/MainWindow.java

```diff
@@ -116,13 +116,17 @@ public int open() {
     Shell shell = getShell();
     Display display = shell.getDisplay();
     while (!shell.isDisposed()) {
-      if (!display.readAndDispatch()) {
-        System.err.println("false");
+      long start = System.nanoTime();
```

Comment:
Maybe you want to use Ben's [Awesome Profiling Counters™](https://github.com/google/gapid/blob/master/gapidapk/android/app/src/main/java/com/google/android/gapid/Counter.java).

That or a test that does profiling?

Reply:
😄 Haha, right. I have used a profiler while working on these changes. This is a little commented out code that when un-commented will take over the UI loop and show how much time is spent on the UI thread in each loop iteration. If it gets large the UI is laggy and if it keeps going, even if nothing is changing in the UI, there's some runaway work being scheduled on the UI thread. Useful for a quick check without the profiler setup/overhead...